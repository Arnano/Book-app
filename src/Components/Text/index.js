import React from 'react';
import PropTypes from 'prop-types';
import { keys } from 'ramda';
import * as colors from '../../Style/colors';
import StyledText from './style';

const Text = ({ children, classes, color, size, weight }) =>
  <StyledText
    className={classes}
    color={color}
    size={size}
    weight={weight} >
      {children}
  </StyledText>

Text.defaultProps = {
  classes: '',
  color: 'darkGrey',
  size: 'fsMedium',
  weight: 'fwMedium'
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  classes: PropTypes.string,
  color: PropTypes.oneOf(keys(colors)),
  size: PropTypes.oneOf(['fsSmall', 'fsMedium', 'fsLarge']).isRequired,
  weight: PropTypes.oneOf(['fwSmall', 'fwMedium', 'fwLarge'])
};

export default Text;
