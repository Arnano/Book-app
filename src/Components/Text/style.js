import styled from 'styled-components';
import { switchProp } from 'styled-tools';
import * as colors from '../../Style/colors';
import { fsSmall, fsMedium, fsLarge, fwSmall, fwMedium, fwLarge } from '../../Style/fonts';

const StyledText = styled.span`
  ${switchProp('size', {
    fsSmall,
    fsMedium,
    fsLarge
  })};

  ${switchProp('weight', {
    fwSmall,
    fwMedium,
    fwLarge
  })};

  color: ${props => colors[props.color]};
`;

export default StyledText;
