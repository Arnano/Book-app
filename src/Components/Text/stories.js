import React from 'react';
import { storiesOf } from '@storybook/react';
import { wwithInfo, withInfo } from '@storybook/addon-info';
import { withKnobs, text, select } from '@storybook/addon-knobs';

import * as colors from '../../Style/colors';
import { keys } from 'lodash';
import Text from './';

const sampleText = `
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
`;


const selectColors = keys(colors);

const selectSize = {
  fsSmall: 'fsSmall',
  fsMedium: 'fsMedium',
  fsLarge: 'fsLarge'
};

const selectWeight = {
  fwSmall: 'fwSmall',
  fwMedium: 'fwMedium',
  fwLarge: 'fwLarge'
};

storiesOf('UI Component', module)
  .add('Text', 
  withInfo({
    text: `
      Text component is used for every small piece of text (i.e. span) accross the application.
      Color, alignment, font-size and font-weight are customizable.
    `
  }) (() =>
    <Text
      children={text('Text: ', sampleText)}
      size={select('Font size: ', selectSize)}
      weight={select('Font weight: ', selectWeight)}
      color={select('Color: ', selectColors)} />
  )
)