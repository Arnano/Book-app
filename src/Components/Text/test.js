import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Text from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Text', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Text size='fsSMall'> Text </Text>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Text size='fsSMall'> Text </Text>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
