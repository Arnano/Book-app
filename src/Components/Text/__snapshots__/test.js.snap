// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text has a valid snapshot 1`] = `
<span
  className="sc-bdVaJa gOaQWy"
  color="darkGrey"
  size="fsSmall"
>
   Text 
</span>
`;
